File: src/interfaces/organization-profile.ts
import { prop, Typegoose, ModelType, InstanceType } from 'typegoose';
import IAddress from './address';

export default class IOrganizationProfile {
	@prop({ required: true })
	schema_version: string;

	@prop()
	mission?: string;

	@prop()
	quote?: string;

	@prop()
	address?: IAddress;

	@prop()
	affiliated_orgs?: Array<string>;

	@prop()
	interests?: Array<string>;
}

File: src/interfaces-internal/organization-profile.go
package interfaces_internal

type IOrganizationProfile struct {
	SchemaVersion  string   `bson:"schema_version"`
	Mission        string   `bson:"mission"`
	Quote          string   `bson:"quote"`
	Address        IAddress `bson:"address"`
	AffiliatedOrgs []string `bson:"affiliated_orgs"`
	Interests      []string `bson:"interests"`
}
